src/gui/properties/peerlistwidget.cpp

```diff
@@ -179,15 +182,17 @@ void PeerListWidget::showPeerListMenu(const QPoint&)
   QAction *act = menu.exec(QCursor::pos());
   if (act == 0) return;
   if (act == addPeerAct) {
-    BitTorrent::PeerAddress addr = PeerAdditionDlg::askForPeerAddress();
-    if (!addr.ip.isNull()) {
-      if (torrent->connectPeer(addr))
-        QMessageBox::information(0, tr("Peer addition"), tr("The peer was added to this torrent."));
-      else
-        QMessageBox::critical(0, tr("Peer addition"), tr("The peer could not be added to this torrent."));
-    }
-    else {
-      qDebug("No peer was added");
+    QList<BitTorrent::PeerAddress> peersList = PeersAdditionDlg::askForPeersEndpoints();
+    if (!peersList.isEmpty()) {
```

Comment:
@glassez True but I think It never goin to happen because the IP/port are valid at this point. I don't know what are the reasons to not be accepted, maybe in the ip filter list? I tried with a lot of weird things and never fails. Any suggestion?